From bcf49db996b9527a23e404229a82ea5290fb76cd Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Tue, 27 Jan 2015 17:11:14 +0800
Subject: [PATCH 0090/5242] MLK-10101-4 usb: add otg_fsm pointer in usb_bus

commit  f881e284f4b63bc8cf9d6adce44f732a9fb85f5d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add otg_fsm pointer in struct of usb_bus for access otg_fsm via bus.
Original way was to put it in usb_otg, then usb host can access otg_fsm
via hcd->usb_phy->otg->fsm, since usb_phy will not be the future direction,
instead phy is preferred, so this way may not work. It's more direct and
simple to put it in usb_bus.

Signed-off-by: Li Jun <jun.li@freescale.com>
(cherry picked from commit c118afccbbb1c78938c649ee72758213c22cf489)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/linux/usb.h |    1 +
 1 file changed, 1 insertion(+)

diff --git a/include/linux/usb.h b/include/linux/usb.h
index 5e49e82..a859501 100644
--- a/include/linux/usb.h
+++ b/include/linux/usb.h
@@ -433,6 +433,7 @@ struct usb_bus {
 					 * Does the host controller use PIO
 					 * for control transfers?
 					 */
+	struct otg_fsm *otg_fsm;	/* usb otg finite state machine */
 	u8 otg_port;			/* 0, or number of OTG/HNP port */
 	unsigned is_b_host:1;		/* true during some HNP roleswitches */
 	unsigned b_hnp_enable:1;	/* OTG: did A-Host enable HNP? */
-- 
1.7.9.5

